packages/smooth_app/lib/pages/scan/continuous_scan_page.dart

```diff
@@ -106,3 +111,130 @@ class ContinuousScanPage extends StatelessWidget {
     await model.refresh();
   }
 }
+
+class CameraRequestButton extends StatefulWidget {
+  @override
+  State<CameraRequestButton> createState() => _CameraRequestButtonState();
+}
+
+class _CameraRequestButtonState extends State<CameraRequestButton> {
+  late PermissionStatus _permissionStatus;
+  late ContinuousScanModel _model;
+  late AppLocalizations _appLocalizations;
+
+  @override
+  Widget build(BuildContext context) {
+    _model = context.watch<ContinuousScanModel>();
+    _permissionStatus = _model.cameraStatus;
+    // We should only be here if we don't have permission.
+    assert(!_permissionStatus.isGranted);
+
+
+    _appLocalizations = AppLocalizations.of(context)!;
+    final ColorScheme colorScheme = Theme.of(context).colorScheme;
+    final ThemeProvider themeProvider = context.watch<ThemeProvider>();
+    final MaterialColor materialColor = SmoothTheme.getMaterialColor(themeProvider);
+    final Color? strongBackgroundColor = SmoothTheme.getColor(
+      colorScheme,
+      materialColor,
+      ColorDestination.SURFACE_BACKGROUND,
+    );
+    final Color? strongForegroundColor = SmoothTheme.getColor(
+      colorScheme,
+      materialColor,
+      ColorDestination.SURFACE_FOREGROUND,
+    );
+
+    return Column(
+      children: <Widget>[
+        const Spacer(flex: 2),
+        Center(
+          child: FloatingActionButton.extended(
+            backgroundColor: strongBackgroundColor,
+            onPressed: (){
+              _requestCameraPermission();
+            },
+            label: Padding(
+              padding: const EdgeInsets.all(8.0),
+              child: Row(
+                children: <Widget>[
+                  Icon(Icons.camera_alt, color: strongForegroundColor),
+                  const SizedBox(width: 10.0),
+                  // TODO(Piinks): Localize this text once settled on
+                   Text(
+                     'Grant camera permission to scan',
+                     style: SmoothTheme.getThemeData(
+                         themeProvider.darkTheme ? Brightness.dark : Brightness.light,
+                         themeProvider.colorTag
+                     ).textTheme.subtitle1!.copyWith(color: strongForegroundColor),
+                   ),
+                ]
+              ),
+            )
+          ),
+        ),
+        const Spacer(),
+      ],
+    );
+  }
+
+  Future<void> _requestCameraPermission() async {
+    PermissionStatus? status;
+    switch (_permissionStatus) {
+      case PermissionStatus.denied:
+      case PermissionStatus.restricted:
+      case PermissionStatus.limited:
+        // Request status
+        status = await Permission.camera.request();
+        break;
+      case PermissionStatus.permanentlyDenied:
+        // We cannot resolve this here, show a dialog directing the user to
+        // system preferences to fix camera permissions.
+        _redirectToSettings();
```

Comment:
[Docs](https://pub.dev/documentation/permission_handler_platform_interface/3.6.1/permission_handler_platform_interface/PermissionStatus-class.html):
> Permission to the requested feature is permanently denied, the permission dialog will not be shown when requesting this permission. The user may still change the permission status in the settings. **Only supported on Android.**

We should probably display something else for iOS